### ISSUE TITLE
Release build for NPM publishing might randomly run in non-release mode

### ISSUE THREAD

**Issue by mourner** [Leaflet/Leaflet#8460]
OK, I figured out why https://unpkg.com/leaflet@1.9.0/dist/leaflet.js has the hash in the version (meaning it was built without `NODE_ENV=release`. It's a tricky one!

According to the [raw job log of v1.9.0 tag build job](https://pipelines.actions.githubusercontent.com/serviceHosts/63923d90-4a1f-44c3-8ec1-bd59c3ff7d7c/_apis/pipelines/1/runs/1868/signedlogcontent/29?urlExpires=2022-09-22T19%3A36%3A45.3572655Z&urlSigningMethod=HMACV1&urlSignature=dhSdPG3oXNKd0kWIbXr4cSWhIvbaXD1%2FD2Td27fejV8%3D), it built with `NODE_ENV=release` successfully. But later scrapped that build because it already found a build saved under the same cache key!

```
Cache hit occurred on the primary key Linux-a7e1bbcb779f181c599086f12555cab851ded25e-build, not saving cache.
```

So, basically it fetched the build that was created on a non-tagged job of the same commit, which was run in non-release mode but saved under the same cache key. To fix this, we'll probably have to update the related cache keys to include the tag name, e.g.:

```yaml
key: ${{ runner.os }}-${{ github.sha }}-${{ github.ref_name }}-build
```

**Comment by jonkoops** [comment]
Ahhhh, yeah that makes a lot of sense! Let me see if I can come up with a fix.
